Add Disable Comment Loading to Data Saver Settings

CThompson01 opened the pull request:
# Feature Description 
Added the ability to disable comment fetching from the data saver settings when viewing the post details screen. This allows for posts that are too long for the preview to be viewed fully without loading comments.

The setting added is shown below.
![image](https://user-images.githubusercontent.com/18427970/208226679-f6bc7ef0-465d-4792-9718-a3d825787942.png)

# Reason for Adding the Feature
This is a feature that I have seen other apps implement and can be helpful for users who need to save data or just don't care about viewing comments. Users will mostly get use out of this feature when viewing posts that are lengthy and too long to be shown fully without viewing the post details screen.

CThompson01 committed:
Add Disable Comments Toggle to Data Saver

Added the ability to disable comment fetching for the data saver settings.

CThompson01 committed:
Merge branch 'Docile-Alligator:master' into master

Docile-Alligator commented:
Thank you. But changes in `PostRecyclerViewAdapter` is not needed since comment loading happens in ViewPostDetailFragment. For the same reason, `ChangeDisableCommentLoadingEvent` should be removed too since we do not have to observe the preference change.

CThompson01 committed:
Remove Disable Comment Event

Remove disable comment event since the preference can only be changed when in the data saver screen. Fixed an issue where the Disable Comment preference on the Data Saver screen would only update when the screen was exited and re-entered.

CThompson01 committed:
Merge branch 'master' of https://github.com/CThompson01/Infinity-For-Reddit
